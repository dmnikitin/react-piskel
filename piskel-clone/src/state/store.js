import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { getFromLocalStorage } from '../helpers/localStorageHandler';
import combined from './reducers/combined';

const localStorageData = getFromLocalStorage();
const store = createStore(combined, localStorageData, composeWithDevTools());

export default store;
