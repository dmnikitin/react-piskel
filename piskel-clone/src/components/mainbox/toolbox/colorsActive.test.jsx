import React from 'react'
import { shallow, mount } from 'enzyme';
import ColorsActive from './colorsActive';
import store from '../../../state/store';

const props = {
  primaryColor: 'black',
  alternativeColor: 'white',
  callback: () => { },
};

jest.mock('../button.jsx', () => 'Button');
const wrapperShallow = shallow(<ColorsActive {...props} />);
const wrapperMount = mount(<ColorsActive {...props} store={store} />);

describe('pensize component', () => {
  it('renders correctly', () => {
    expect(wrapperShallow).toMatchSnapshot();
  });
  it("displays initial div", () => {
    expect(wrapperMount.find("div")).toHaveLength(3);
  });
  it('displays proper amount of buttons', () => {
    expect(wrapperMount.find('.toolbox-colors-active').children()).toHaveLength(3);
  });
});
