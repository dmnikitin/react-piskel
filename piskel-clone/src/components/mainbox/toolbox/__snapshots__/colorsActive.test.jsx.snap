// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pensize component renders correctly 1`] = `
<div
  className="toolbox-colors-active"
>
  <div
    className="toolbox-colors-active__primary"
    style={
      Object {
        "background": "black",
      }
    }
  />
  <Button
    callback={[Function]}
    data="swap"
    icon="exchange"
  />
  <div
    className="toolbox-colors-active__alternative"
    style={
      Object {
        "background": "white",
      }
    }
  />
</div>
`;
